manticore/utils/rlp.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+import collections.abc
+
+def int_to_bytes(n):
+    return n.to_bytes((n.bit_length() + 7) // 8, 'big')
+
+def rlp_encode(item):
+    if item is None or item == 0:
+        return b'\x80'
+    elif isinstance(item, str):
+        return rlp_encode(item.encode('utf8'))
+    elif isinstance(item, bytearray) or isinstance(item, bytes):
+        if len(item) == 1 and item[0] < 0x80:
+            return item
+        else:
+            return encode_length(len(item), 0x80) + item
+    elif isinstance(item, collections.abc.Sequence):
+        output = b''.join(map(rlp_encode, item))
+        return encode_length(len(output), 0xC0) + output
+    elif isinstance(item, int) or isinstance(item, long):
+        return rlp_encode(int_to_bytes(item))
+    else:
+        raise Exception("Cannot encode object of type %s" % type(item).__name__)
+
+def encode_length(length, offset):
+    if length < 56:
+         return int_to_bytes(length + offset)
+    elif length < 256**8:
+         binary = to_binary(length)
+         return int_to_bytes(len(binary) + offset + 55) + binary
+    else:
+         raise Exception("length must be less than %d" % 256**8)
+
+def to_binary(x):
+    if x == 0:
+        return b''
+    else: 
+        return to_binary(int(x / 256)) + int_to_bytes(x % 256)
+
+if __name__ == "__main__":
+    import sys
+    passed = True
+    def to_list(encoded):
+        ret = []
+        for c in encoded:
+            if c >= ord(' ') and c <= ord('~'):
+                ret.append(chr(c))
+            else:
+                ret.append(c)
+        return ret
+    def to_int(exp):
+        if isinstance(exp, str):
+            return ord(exp)
+        else:
+            return exp
+    LOREM_IPSUM = "Lorem ipsum dolor sit amet, consectetur adipisicing elit"
+    for item, expected in (
```

Comment:
Those tests could be written using [doctest](https://docs.python.org/3/library/doctest.html)